### ISSUE TITLE
Start at boot on ICS, Android 4.0: Add missing RECEIVE_BOOT_COMPLETED permission to the manifest.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#267]
```


What steps will reproduce the problem?
1. install Android ICS Rom :)
2. install GTalkSMS
3. Reboot the phone

What is the expected output?
I would like to see to restart GTalkSMS at boot.

What do you see instead?
It wasn't started.


Additional informations:
Phone model: SE Xperia Ray Official ICS 4.0.3

Solution:
You have to add the RECEIVE_BOOT_COMPLETED permission to the manifest.

Thank you!

```

Original issue reported on code.google.com by `d4n...@gmail.com` on 22 Apr 2012 at 6:14

**Comment by GoogleCodeExporter** [comment]
```
I've tested this permission modification on my device, it's working ;)
```

Original comment by `d4n...@gmail.com` on 22 Apr 2012 at 6:57

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the report :)
```

Original comment by `fschm...@gmail.com` on 23 Apr 2012 at 7:18
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 23 Apr 2012 at 7:18
- Changed title: **Start at boot on ICS, Android 4.0: Add missing RECEIVE_BOOT_COMPLETED permission to the manifest.**

**Comment by GoogleCodeExporter** [comment]
```
Fixed with 
http://code.google.com/p/gtalksms/source/detail?r=5f8273042c4c87acb2b9e52652c191
b467b5306c
```

Original comment by `fschm...@gmail.com` on 2 May 2012 at 7:47
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
I have exactly the same problem.
I am wondering is the solution only for rooted phones?
Since I have official 4.0.3 ICS on GS2. And I haven't added any permissions to 
manifest and not familiar with the process.
Would you please comment on this matter ?

Thanks in advance for your help
```

Original comment by `Mehrad.A...@gmail.com` on 4 Jun 2012 at 11:57

**Comment by GoogleCodeExporter** [comment]
```
This solution works on all phone under ICS. No user manipulation is required.

Can you try the last beta version with the fix in the download section:
https://code.google.com/p/gtalksms/downloads/detail?name=GTalkSMS-Holo-Beta1-sig
ned-aligned.apk&can=2&q=



```

Original comment by `Florent....@gmail.com` on 4 Jun 2012 at 12:03

**Comment by GoogleCodeExporter** [comment]
```
Form your answer I understand that adding permissions to manifest isn't limited 
for rooted ROMs and can be done in official ROM.
I would appreciate it if you can give me some simple clews to look for then I 
can figure out how to start and do the process. 
If not I would wait for the final version of the beta you mentioned to get 
released then I can update.

Thanks for your quick reply
```

Original comment by `Mehrad.A...@gmail.com` on 4 Jun 2012 at 12:40

**Comment by GoogleCodeExporter** [comment]
```
The only way to add a permission on a stock ROM is to rebuild the application.
The next release will be available soon.


```

Original comment by `Florent....@gmail.com` on 4 Jun 2012 at 2:14

**Comment by GoogleCodeExporter** [comment]
```
I get it now. Thanks heaps.
```

Original comment by `Mehrad.A...@gmail.com` on 4 Jun 2012 at 11:23

**Comment by GoogleCodeExporter** [comment]
```
Fixed with GTalkSMS v4.0 and released on market
```

Original comment by `Florent....@gmail.com` on 28 Jul 2012 at 4:20
- Changed state: **Deployed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed with GTalkSMS v4.0 and released on market
```

Original comment by `Florent....@gmail.com` on 28 Jul 2012 at 4:20
